testsuite/MDAnalysisTests/formats/test_libmdaxdr.py

```diff
@@ -21,376 +21,543 @@
+def test_seek_overflow(xdrfile, fname):
+    with xdrfile(fname) as f:
+        with pytest.raises(OverflowError):
+            f._bytes_seek(2**65)
 
 
-def test_xtc():
-    f = XTCFile(XTC_single_frame)
-    assert_equal(f.n_atoms, 10)
-    xyz, box, step, time, prec = f.read()
-    # xtc only saves with 3 decimal places precision
-    assert_array_almost_equal(xyz.flat, np.arange(30), decimal=3)
-    assert_array_almost_equal(box, np.eye(3) * 20, decimal=3)
-    assert_equal(step, 0)
-    assert_equal(time, 10.0)
-    assert_equal(prec, 1000.0)
+@pytest.mark.parametrize("xdrfile, fname, offsets",
+                         ((XTCFile, XTC_multi_frame, XTC_OFFSETS),
+                          (TRRFile, TRR_multi_frame, TRR_OFFSETS)))
+def test_seek_tell_all(xdrfile, fname, offsets):
+    with xdrfile(fname) as f:
+        for frame, offset in enumerate(offsets):
+            f.seek(frame)
+            assert f._bytes_tell() == offset
 
 
-def test_read_multi_frame_xtc():
-    with XTCFile(XTC_multi_frame) as f:
-        f.seek(5)
-        assert_equal(f.tell(), 5)
-        assert_equal(len(f), 10)
-        assert_equal(f.tell(), 5)
-        f.seek(0)
-        assert_equal(f.tell(), 0)
+@pytest.mark.parametrize("xdrfile, fname, offsets",
+                         ((XTCFile, XTC_multi_frame, XTC_OFFSETS),
+                          (TRRFile, TRR_multi_frame, TRR_OFFSETS)))
+def test_seek_tell_all_bytes(xdrfile, fname, offsets):
+    with xdrfile(fname) as f:
+        for offset in offsets:
+            f._bytes_seek(offset)
+            assert f._bytes_tell() == offset
+
+
+@pytest.mark.parametrize("xdrfile, fname", ((XTCFile, XTC_multi_frame),
+                                            (TRRFile, TRR_multi_frame)))
+def test_seek_tell_largefile(xdrfile, fname):
+    # Seeking/telling can be done on offsets larger than the file.
+    # Filesize won't change unless a write is done at the offset.
+    # We attempt to go just beyond a 4GB limit.
+    big_offset = 2**32 + 1000
+    with xdrfile(fname) as f:
+        # It should be safe to leave the read head so far away
+        # because the file is open read-only.
+        f._bytes_seek(big_offset)
+        assert f._bytes_tell() == big_offset
+
+
+def test_read_multi_frame_xtc(xtc):
+    xtc.seek(5)
+    assert xtc.tell() == 5
+    assert len(xtc) == 10
+    assert xtc.tell() == 5
+    xtc.seek(0)
+    assert xtc.tell() == 0
+    ones = np.ones(30).reshape(10, 3)
+    box_compare = np.eye(3) * 20
+    for i, frame in enumerate(xtc):
+        xyz, box, step, time, prec = frame
+        assert_array_almost_equal(xyz, ones * i, decimal=3)
+        assert_array_almost_equal(box, box_compare, decimal=3)
+        assert step == i
+        assert time == i * 0.5
+        assert prec == 1000.0
+
+
+@pytest.mark.parametrize(
+    'dtype', (np.int32, np.int64, np.float32, np.float64, int, float))
+def test_write_xtc_dtype(tmpdir, dtype, xtc):
+    outname = str(tmpdir.join('foo.xtc'))
```

Comment:
@richardjgowers This is another option to use `tmpdir`. It avoids the extra indentation.